rust/sdk/src/client.rs

```diff
@@ -457,6 +458,684 @@ pub(crate) fn to_lines(stream: impl Stream<Item = Result<Bytes, reqwest::Error>>
         .flatten()
 }
 
+/// Request builder for event publishing.
+///
+/// Warning: [`Publish`] implements the [`Future`] trait, as such it can be polled.
+/// Calling _any_ [`Publish`] function after polling will result in a panic!
+pub enum Publish<'a> {
+    Initial { client: &'a Ax, request: PublishRequest },
+    Pending(BoxFuture<'a, anyhow::Result<PublishResponse>>),
+    Void,
+}
+
+impl<'a> Publish<'a> {
+    fn new(client: &'a Ax) -> Self {
+        Self::Initial {
+            client,
+            request: PublishRequest { data: vec![] },
+        }
+    }
+
+    /// Add an event.
+    ///
+    /// Subsequent calls to this function will not remove previous events.
```

Comment:
This is close to a double negation. How about “This adds the given event to the list of events that will be emitted once this publishing request is submitted by using `.await` on it.”